calculator_test.go

```diff
@@ -1,24 +1,173 @@
+		{-51, []float64{-51}, 0, "Subtracting a negative number from itself yields zero"},
+		{4323, []float64{12, 11, 39}, 4261, "Subtracting 3 positive number from a much larger positive number yields a positive number"},
+		{11.2, []float64{89.8}, -78.6, "Subtracting a positive number from a smaller positive number yields a negative number"},
+		{-100, []float64{-111}, 11, "Subtracting a negative number from a larger negative number yields a positive number"},
+		{0, []float64{6, 6, 15}, -27, "Subtracting multiple positive number from zero yields a negative number"},
+		{0, []float64{-22}, 22, "Subtracting a negative number from zero yields a positive number"},
+	}
+	for _, c := range testCases {
+		got := Subtract(c.a, c.b...)
+		if c.want != got {
+			t.Errorf("%s: want %f, got %f", c.name, c.want, got)
+		}
+	}
+}
+
+func TestMultiply(t *testing.T) {
+	t.Parallel()
+	var testCases = []struct {
+		a    float64
+		b    []float64
+		want float64
+		name string
+	}{
+		{1, []float64{1}, 1, "Multiplying one by one always yields one"},
+		{3999191, []float64{1, 1, 1}, 3999191, "Multiplying a number by one any number of times yields that number"},
+		{3, []float64{3}, 9, "Multiplying a number by itself equals that number squared"},
+		{-10, []float64{-10}, 100, "Multiplying two negative number yields a positive number"},
+		{-5, []float64{-12, -67}, -4020, "Multiplying three negative number yields a negative number"},
+		{12, []float64{-1.5}, -18, "Multiplying a negative number by a positive number yields a negative number"},
+		{43, []float64{0}, 0, "Multiplying any number by zero yields zero"},
+	}
+	for _, c := range testCases {
+		got := Multiply(c.a, c.b...)
+		if c.want != got {
+			t.Errorf("%s: want %f, got %f", c.name, c.want, got)
+		}
+	}
+}
+
+func TestDivide(t *testing.T) {
+	t.Parallel()
+	var testCases = []struct {
+		a             float64
+		b             []float64
+		want          float64
+		errorExpected bool
+		name          string
+	}{
+		{10, []float64{1}, 10, false, "Dividing a number by 1 yields that number"},
+		{99, []float64{1, 1, 1}, 99, false, "Dividing a number by one multiple times yields that same number"},
+		{312, []float64{0}, 0, true, "Dividing a number by zero returns an error"},
+		{50, []float64{1, 5, 0, 200}, 0, true, "Dividing a number by multiple numbers with zero as one of them returns an error"},
+		{0, []float64{0}, 0, true, "Dividing a zero by zero returns an error"},
+		{1, []float64{2, 2, 2}, 0.125, false, "Dividing a number by multiple larger numbers yields a fraction"},
+		{100, []float64{0.1, 10}, 100, false, "Dividing a number by a large number and a fraction"},
+		{100, []float64{10000000}, 0.00001, false, "Dividing a number by a much larger number yields a small fraction"},
+		{100, []float64{10000000}, 0.00001, false, "Dividing a number by a much larger number yields a small fraction"},
+	}
+	for _, c := range testCases {
+		got, err := Divide(c.a, c.b...)
+		if c.errorExpected && err == nil {
```

Comment:
And we can catch both 'unexpected error status' cases in one go:

```go
errReceived := err != nil
if c.errExpected != errReceived {
	t.Fatalf("%s: unexpected error status %v", c.name, err)
}
```

It's `Fatal` because if the error status doesn't match, we _must not_ compare the data value against expectation: it's by definition invalid. Even attempting to dereference it could panic, if it were a pointer, for example.

Reply:
Thanks for that! Makes sense and done.